### PR TITLE
Add motivation to tree readme

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Ad some details to tree's readme covering its motivation, and alternatives.

## Reviewer Guidance

Needs proof reading/editing as well as high level review of structure, semantics and general ideas.

### CODE REVIEW

File: packages/dds/tree/README.md [hunk 1]
```diff
@@ -1,3 +1,59 @@
...
+
+## Why not a tree made out of existing DDS implementations and how this relates to the Fluid Framework itself?
+
+Currently existing DDS implementations can not support cross DDS transactions.
+For example, moving part of a sequence from one sequence DDS to another can no be done transactionally meaning if the source of the move conflicts, the destination half can't be updated or aborted if its in a different DDS.
+Cross DDS moves also currently can't be as efficient as moves withing a single DDS, and there isn't a good way to do cross DDS history or branching without major framework changes.
+There are also some significant per DDS performance and storage costs that make this approach much more costly than using a single DDS.
+
+One way to think about this new tree DDS is to try and mix some of the fluid-framework features (like the ability to checkout a subset of the data) with features from DDSs (ex: lower overhead per item, efficient moves of sub-sequences, transactions).
+If this effort is successful, it might revel some improved abstractions for modularizing hierarchical collaborative data-structures (perhaps "field kinds"),
```
Comment (yann-achard-MS) [inline packages/dds/tree/README.md:55]: `revel` => `reveal`

File: packages/dds/tree/README.md [hunk 2]
```diff
@@ -1,3 +1,59 @@
...
+
+## Whats missing from existing DDSs?
+
+`directory` and `map` can not provide merge resolution that guarantees well-formedness of trees while supporting the desired editing APIs (like subsequence move),
+and are missing (can can not be practically extended to have) efficient ways to handle large data ort schema.
+
+`sequence` does not capture the hierarchy or schema, and also does not handle partial checkouts.
+Additionally its actual merge resolution leaves some things to be desired in some cases which `tree` aims to improve on.
+
+`expirmental/tree` does not have a built in schema system reducing the data available to make semantically high quality merges.
```
Comment (justus-camp) [inline packages/dds/tree/README.md:39]: `expirmental/tree` => `experimental/tree`

File: packages/dds/tree/README.md [hunk 3]
```diff
@@ -1,3 +1,59 @@
...
+It also does merge resolution in a way that does not scale to not having the whole tree in memory due to it being based entirely on node identifiers,
```
Comment (justus-camp) [inline packages/dds/tree/README.md:40]: Don't think this sentence reads very well.

Maybe `It also doesn't do merge resolution in a way that doesn't require having the whole tree in memory due to it being based entirely on node identifiers (including constraints within transactions that can't be verified without reading large parts of the tree).`

File: packages/dds/tree/README.md [hunk 4]
```diff
@@ -1,3 +1,59 @@
...
+(including constraints within transactions that can't be verified without reading large parts of the tree).
+
+`expirmental/PropertyDDS` currently does not have as high quality merge logic as desired, currently not even supporting efficient moves.
+Much of what is desired is theoretically possible as additional feature work on `PropertyDDS`,
+but it was decided that it makes more sense to build up this more featureful DDS from scratch leveraging the learnings from `PropertyDDS` and `expirmental/tree`.
```
Comment (justus-camp) [inline packages/dds/tree/README.md:45]: `expirmental/tree` => `experimental/tree`

File: packages/dds/tree/README.md [hunk 5]
```diff
@@ -1,3 +1,59 @@
...
+
+## Why not a tree made out of existing DDS implementations and how this relates to the Fluid Framework itself?
+
+Currently existing DDS implementations can not support cross DDS transactions.
+For example, moving part of a sequence from one sequence DDS to another can no be done transactionally meaning if the source of the move conflicts, the destination half can't be updated or aborted if its in a different DDS.
```
Comment (yann-achard-MS) [inline packages/dds/tree/README.md:50]:
```suggestion
For example, moving part of a sequence from one sequence DDS to another cannot be done transactionally, meaning if the source of the move conflicts, the destination half can't be updated or aborted if it's in a different DDS.
```

**Review by yann-achard-MS — APPROVED** [review]

**Review by yann-achard-MS — APPROVED** [review]

**Pull request merged**